### PR TITLE
Added ARIA labels

### PR DESCRIPTION
**Pull request by hborghols**
### This PR:

Relates to #11 

**Bugfixes/Changed internals** 🎈
- Adds ARIA labels according to https://www.w3.org/TR/wai-aria-practices where necessary.
- Adds functionality to close `Modal`/`Popover` with escape key

**Checklist** 🛡
- [x] I have exported my addition from `src/index.ts` (check if not applicable).
- [x] Appropriate tests have been added for my functionality (check if not applicable).
- [x] A designer has seen and approved my changes (tag `@LuukHorsmans` for a design review when applicable).
- [x] I have tested my addition in all supported browsers and for responsiveness (Chrome, Firefox, Safari, Edge, IE11 and mobile browsers).

### CODE REVIEW

File: src/components/Select/index.tsx [hunk 1]
```diff
@@ -197,12 +197,15 @@ class Select<GenericOptionType extends OptionBaseType> extends Component<PropsTy
...
                 onBlur={this.handleBlur}
                 tabIndex={this.props.disabled ? -1 : 0}
+                role="search"
             >
                 <StyledInput
                     open={this.state.isOpen}
                     hasFocus={this.state.hasFocus}
                     disabled={!this.props.disabled ? false : this.props.disabled}
                     ref={this.inputWrapperRef}
+                    role="button"
```
Comment (woutsluiter) [inline src/components/Select/index.tsx]: This whole "form input" seems to be made out of divs. Not really sure if this is in scope of the story, but are there any other a11y gains here? Like `Radio`, `aria-checked` and `aria-labelledby`?

File: src/components/Popover/index.tsx [hunk 2]
```diff
@@ -106,7 +116,12 @@ class Popover extends Component<PropsType, StateType> {
                                     if (ref) this.anchorRef = ref;
                                 }}
                             >
-                                <PopoverAnchor ref={ref} stretch={this.props.stretch}>
+                                <PopoverAnchor
+                                    ref={ref}
+                                    stretch={this.props.stretch}
+                                    role="button"
```
[the remark this answers is not in the document]
Comment (hborghols) [reply src/components/Popover/index.tsx]: The tooltip ARIA standard for the tooltip pattern is not final yet (https://www.w3.org/TR/wai-aria-practices/#tooltip) so I won't add `Tooltip` as a role for now.

File: src/components/Modal/index.tsx [hunk 3]
```diff
@@ -43,12 +43,20 @@ class Modal extends Component<PropsType> {
         }
     };
 
+    public handleKeyDown = (event: KeyboardEvent): void => {
+        if (this.props.onClose !== undefined && (event.key === 'Escape' || event.key === 'Esc')) {
```
Comment (woutsluiter) [inline src/components/Modal/index.tsx]: 🙌

File: src/components/Select/index.tsx (hunk 1 above)
Comment (hborghols) [reply src/components/Select/index.tsx]: Updated this according to the `combobox` guidelines

**Pull request merged**